src/source/src/parser/mod.rs

```diff
@@ -261,7 +262,11 @@ pub trait SourceParser: Send + Sync + Debug + 'static {
     /// # Returns
     ///
     /// A [`WriteGuard`] to ensure that at least one record was appended or error occurred.
-    fn parse(&self, payload: &[u8], writer: SourceStreamChunkRowWriter<'_>) -> Result<WriteGuard>;
+    async fn parse(
+        &self,
+        payload: &[u8],
+        writer: SourceStreamChunkRowWriter<'_>,
+    ) -> Result<WriteGuard>;
```

Comment:
I see no necessity to declare parse as `async`. `parse` is in our critical path. async would slow it down.

Reply:
The parser has a `ConfluentSchemaResolver `, which will make async http request when encountering a new schema. If the `parse` method is not async, we need to use a sync http lib to write the code. In terms of performance, the 
implementation of `parse` does not contain blocking operations which would cause task switch, except for getting the schema, I don't think it will harm performance.

Reply:
After discussing with @BugenZhao , async itself has no overhead, but `async_trait` generates `Box<dyn Future>`, which may harm performance, so it can be changed to static dispatch here, like [store](https://github.com/risingwavelabs/risingwave/blob/main/src/storage/src/store.rs).

Reply:
Use GAT + TAIT to resolve it in a future PR.